File: Parte4/ej4a.m
% Problema 4a: Convertir 123.51 a binario y hexadecimal
num = 123.51;
binario_entero = dec2bin(floor(num)); % Parte entera en binario
binario_fraccion = ''; 
frac = num - floor(num);
while frac > 0 && length(binario_fraccion) < 10
    frac = frac * 2;
    bit = floor(frac);
    binario_fraccion = strcat(binario_fraccion, num2str(bit));
    frac = frac - bit;
end
binario = strcat(binario_entero, '.', binario_fraccion);
hexadecimal = dec2hex(floor(num));

% Mostrar resultados
disp(['Binario de 123.51: ', binario]);
disp(['Hexadecimal de 123.51: ', hexadecimal]);

File: Parte4/ej4a.c
#include <stdio.h>
#include <math.h>

void decimalABinario(double num) {
    int parteEntera = (int)num;
    double parteFracc = num - parteEntera;
    
    char binarioEntero[20];
    itoa(parteEntera, binarioEntero, 2); // Convierte parte entera a binario
    
    printf("Parte entera en binario: %s.", binarioEntero);
    
    printf(" Parte fraccionaria en binario: ");
    for (int i = 0; i < 10 && parteFracc > 0; i++) {
        parteFracc *= 2;
        if (parteFracc >= 1) {
            printf("1");
            parteFracc -= 1;
        } else {
            printf("0");
        }
    }
    printf("\n");
}

int main() {
    double num = 123.51;
    
    printf("Número en hexadecimal: %X\n", (int)num); // Convierte parte entera a hexadecimal
    decimalABinario(num);
    
    return 0;
}
